hayeon/programmers-2/괄호-회전하기.js

```diff
@@ -0,0 +1,41 @@
+function solution(s) {
+  const bracketsList = s.split("");
+  const bracketObject = {
+    "[": "]",
+    "{": "}",
+    "(": ")",
+  };
+
+  let answer = 0;
+
+  for (let i = 0; i < bracketsList.length; i++) {
+    bracketsList.push(bracketsList.shift());
+
+    const temp = [];
+    let isCheck = true;
+
+    for (let j = 0; j < bracketsList.length; j++) {
+      const string = bracketsList[j];
+
+      if (!bracketObject[string]) {
+        if (bracketObject[temp.pop()] !== string) {
+          isCheck = false;
+          break;
+        }
+      } else {
+        temp.push(string);
+      }
+    }
+
+    if (temp.length === 0 && isCheck) answer++;
+  }
+
+  return answer;
+}
```

Comment:
```suggestion
function solution(s) {
  const bracketsList = s.split("");
  const bracketObject = {
    "[": "]",
    "{": "}",
    "(": ")",
  };

  let answer = 0;

  for (let i = 0; i < bracketsList.length; i++) {
    const stack = [];
    let isCheck = true;
      
    bracketsList.push(bracketsList.shift());

    for (let j = 0; j < bracketsList.length; j++) {
      const bucket = bracketsList[j];

      if (!bracketObject[bucket]) {
        if (bracketObject[stack.pop()] !== bucket) {
          isCheck = false;
          break;
        }
      } else {
        stack.push(bucket);
      }
    }

    if (stack.length === 0 && isCheck) answer++;
  }

  return answer;
}
```
- 풀이 잘봤습니다 고생하셨습니다~
- stack을 사용한 만큼 temp는 stack으로 네이밍 변경을 하였습니다.
- 또한 string이라는 네이밍보다는 bucket 으로 네이밍 변경하였습니다.
- 네이밍에 좀 더 신경쓰면 좋을 것 같습니다~ 🙇‍♂️